retrieve and revert subclass methods

ddybvig-perf opened the pull request:
This implements the retrieve and revert methods for all four CredentialConfig subclasses. I wrote tests for good input on LDAP and UMAPI as well as bad input on LDAP. I could include a test for invalid input on UMAPI as well as test for the other subclasses if needed.

ddybvig-perf committed:
renamed subclass methods to retrieve instead of fetch. commented out existing implementation for testing purposes. started on two tests

ddybvig-perf committed:
renamed last "fetch" method to "retrieve." finished valid ldap test then the actual method implementation. had to be different from the example since retrieve_key and revert_key both return strings now and not dicts

ddybvig-perf committed:
added test or invalid input on ldap. wrote test for valid on umapi. need to figure out the nested keys

ddybvig-perf committed:
corrected implementation for setting nested keys in the retrieved/reverted dicts. finished methods for okta and console config subclasses

ddybvig-perf committed:
refactored revert subclass methods to return None and removed assertions from tests that were redundant with retrieve

ddybvig-perf commented:
python 2.7 build fails. 3.6 passes

vossen-adobe commented on user_sync/credentials.py:
```diff
@@ -264,14 +260,20 @@ def store(self):
+        self.revert_key(['enterprise', 'api_key'])
+        self.revert_key(['enterprise', 'client_secret'])
+        self.revert_key(['enterprise', 'tech_acct'])
+        self.save()
```
Let's return the fetched credentials just like with retrieve

vossen-adobe reviewed (changes requested):
Let's return a dict of values from revert too, so we know what was reverted - for all the classes

ddybvig-perf committed:
modify all revert subclass methods to return a dict of reverted keys

ddybvig-perf committed:
added documentation updates

vossen-adobe committed:
formatting

vossen-adobe committed:
Merge branch 'feature/credential-manager' into dale-its-credman-time

ddybvig-perf committed:
had to import os in tests. valid and invalid input tests for credman retrieve and revert methods. method implementations for credman retrieve and revert methods

ddybvig-perf committed:
change the check for unstored creds in app (can't check for the whole dict being None - have to check individual values instead. small change to revert in app just for variable naming consistency across methods

ddybvig-perf committed:
looping over the retrieved creds dict to check for None values
